client.go

```diff
@@ -66,6 +66,15 @@ var (
 	bufPool           = &sync.Pool{New: func() interface{} { return &bytes.Buffer{} }}
 )
 
+type (
```

Comment:
@mloberg Can you please add godoc for the exported types?

Reply:
added